services/src/machine_learning/xgb_config_builder.rs

```diff
@@ -0,0 +1,1057 @@
+        let mut lp = LearningTaskParametersBuilder::default();
+        let mut bp = BoosterParametersBuilder::default();
+
+        // iterate over all keys of the map
+        // TODO: error handling
+        for (key, value) in map {
+            match key.as_str() {
+                "booster" => {
+                    let bt = BoosterType::from(value.as_str());
+                    bp.booster_type(bt);
+                }
+                "eta" => {
+                    tp.eta(value.parse::<f32>().unwrap());
+                }
+                "hist" | "tree_method" => {
+                    tp.tree_method(TreeMethod::from(value.as_str()));
+                }
+                "process_type" => {
+                    tp.process_type(ProcessType::from(value.as_str()));
+                }
+                "max_depth" => {
+                    tp.max_depth(value.parse::<u32>().unwrap());
+                }
+                "objective" => {
+                    lp.objective(Objective::from(value.as_str()));
+                }
+                "verbosity" => {
+                    bp.verbose(value.parse::<bool>().unwrap());
+                }
+                "colsample_bylevel" => {
+                    tp.colsample_bylevel(value.parse::<f32>().unwrap());
+                }
+                "colsample_bytree" => {
+                    tp.colsample_bytree(value.parse::<f32>().unwrap());
+                }
+                "colsample_bynode" => {
+                    tp.colsample_bynode(value.parse::<f32>().unwrap());
+                }
+                "sketch_eps" => {
+                    tp.sketch_eps(value.parse::<f32>().unwrap());
+                }
+                "max_bin" => {
+                    tp.max_bin(value.parse::<u32>().unwrap());
+                }
+                "max_delta_step" => {
+                    tp.max_delta_step(value.parse::<f32>().unwrap());
+                }
+
+                "num_parallel_tree" => {
+                    tp.num_parallel_tree(value.parse::<u32>().unwrap());
+                }
+                "lambda" => {
+                    linp.lambda(value.parse::<f32>().unwrap());
+                }
+                "alpha" => {
+                    linp.alpha(value.parse::<f32>().unwrap());
+                }
+                "predictor" => {
+                    tp.predictor(Predictor::from(value.as_str()));
+                }
+                "grow_policy" => {
+                    tp.grow_policy(GrowPolicy::from(value.as_str()));
+                }
+                "max_leaves" => {
+                    tp.max_leaves(value.parse::<u32>().unwrap());
+                }
+                "refresh_leaf" => {
+                    let refresh = match value.as_str() {
+                        "1" | "true" => true,
+                        "0" | "false" => false,
+                        //TODO: beter error handling
+                        _ => panic!("Invalid value for refresh_leaf: {value}"),
```

Comment:
Implement TryFrom instead of From and return an error

Reply:
see: https://github.com/geo-engine/geoengine/pull/746#discussion_r1252302470